packages/ActionBar/src/hooks/useSort/useSort.js

```diff
@@ -0,0 +1,83 @@
+import React from "react";
+import nanoid from "nanoid";
+import sort from "./sort";
+
+function sortData({ sortedFields, columns, data }) {
+  let sortedData = data;
+  if (sortedFields.length > 0) {
+    sortedFields.forEach(field => {
+      sortedData = sort({
+        data,
+        columnId: field.columnId,
+        direction: field.direction,
+        columnType: columns.find(column => field.columnId === column.id).type,
+        momentParsingFormat: columns.find(column => field.columnId === column.id).momentParsingFormat,
+      });
+    });
+  }
+  return sortedData;
+}
+
+export default function useSort({ data = null, columns }) {
+  const [sortedFields, setSortedFields] = React.useState([]);
+  const [sortedData, setSortedData] = React.useState(data);
+  const [appliedNumber, setAppliedNumber] = React.useState(0);
+
+  const handleAddItem = React.useCallback(() => {
+    setSortedFields(prevFields => [
+      ...prevFields,
+      {
+        id: nanoid(),
+        columnId: columns[0].id,
+        direction: "ASCEND",
+      },
+    ]);
+  }, [columns]);
+
+  const handleDeleteItem = React.useCallback(
+    deletedId => {
+      setSortedFields(prevFields => [...prevFields].filter(filter => filter.id !== deletedId));
```

Comment:
There is no need to do `[...prevFields]`, because `filter` will return a new array anyway.